controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -530,6 +519,57 @@ private static void writeToSegmentTableAndIndex(int startingSegmentNumber, int n
                 );
     }
 
+    /**
+     * This method takes a reference epoch, seals the active epoch, creates a duplicate entry for referenced epoch.
```

Comment:
You might want to add some motivation for this method in this description. Map it back to rolling txns and why we need to duplicate epochs.